src/capi/azext_capi/custom.py

```diff
@@ -136,21 +107,61 @@ def _install_capz_components(cmd):
             output = subprocess.check_output(command, universal_newlines=True, stderr=stderr)
             logger.info("%s returned:\n%s", " ".join(command), output)
         except subprocess.CalledProcessError as err:
-            raise UnclassifiedUserFault("Can't locate a Kubernetes cluster") from err
+            raise UnclassifiedUserFault("Couldn't install capz components in current cluster") from err
 
 
 def create_management_cluster(cmd, yes=False):
     check_prereqs(cmd)
     msg = 'Do you want to initialize Cluster API on the current cluster?'
     if not yes and not prompt_y_n(msg, default="n"):
-        return
-
-    command = ["clusterctl", "init", "--infrastructure", "azure"]
+        if not create_new_management_cluster(cmd):
+            return
     try:
-        output = subprocess.check_output(command, universal_newlines=True)
-        logger.info("%s returned:\n%s", " ".join(command), output)
+        _find_default_cluster()
     except subprocess.CalledProcessError as err:
-        raise UnclassifiedUserFault("Can't locate a Kubernetes cluster") from err
+        raise UnclassifiedUserFault("Can't locate a kubernetes cluster") from err
+    set_azure_identity_secret_env_vars()
+    _create_azure_identity_secret(cmd)
+    _install_capz_components(cmd)
+
+
+def create_new_management_cluster(cmd, prompt=True, pre_prompt_text=None):
+    choices = ["azure - a management cluster in the Azure cloud",
+               "local - a local Docker container-based management cluster",
+               "exit - don't create a management cluster"]
+
+    if prompt:
+        prompt = pre_prompt_text if pre_prompt_text else ""
+        prompt += """
+Where do you want to create a management cluster?
+"""
+        choice_index = prompt_choice_list(prompt, choices)
+    else:
+        choice_index = 0
+    cluster_name = "capi-manager"
+    if choice_index == 0:
+        command = ["az", "group", "create", "-l", "southcentralus", "--name", cluster_name]
```

Comment:
this isn't a change from this PR but just noticed this as I was reading through, perhaps the location shouldn't be hardcoded. Just something to follow up on later.